google/cloud/examples/gcs2cbt.cc

```diff
@@ -193,33 +137,11 @@ int main(int argc, char* argv[]) try {
     return 1;
   }();
 
-  // The size of the circular buffer. It is tempting to make it larger, but each
-  // element in the circular buffer can be a few MiB in size. We limit this to
-  // the number of threads, once it becomes that large all threads will have
-  // work to do after they finish their current work.
-  int const buffer_size = static_cast<int>(thread_pool_size);
-
-  // Create a circular buffer to communicate between the main thread that reads
-  // the file and the threads that upload the parsed lines to Cloud Bigtable.
-  CircularBuffer buffer(buffer_size);
-  // Then create a few threads, each one of which pulls mutations out of the
-  // circular buffer and then applies the mutation to the table.
-  auto read_buffer = [&buffer](cbt::Table table, int report_progress_rate) {
-    long count = 0;
-    cbt::BulkMutation mutation;
-    while (buffer.Pop(mutation)) {
-      table.BulkApply(std::move(mutation));
-      if (++count % report_progress_rate == 0) {
-        std::cout << '+' << std::flush;
-      }
-    }
-  };
-
   std::cout << "Starting " << thread_pool_size << " workers ..." << std::flush;
-  std::vector<std::future<void>> workers;
-  for (std::size_t i = 0; i != thread_pool_size; ++i) {
-    workers.push_back(std::async(std::launch::async, read_buffer, table,
-                                 report_worker_progress_rate));
+  google::cloud::bigtable::CompletionQueue cq;
+  std::vector<std::thread> thread_pool;
+  for (int i = 0; i < static_cast<int>(thread_pool_size); ++i) {
```

Comment:
I suggest declaring `i` to be std::size_t and avoid the cast.

Reply:
Done